### DISCUSSION TITLE
Generating global features using FakeDataset

### DISCUSSION THREAD

**Discussion by arunppsg** [pyg-team/pytorch_geometric#4067] [Ideas]
Some graphs can have global level features which can be used to represent overall graph level properties. Like, in case of crystals graph in material science, we can have temperature etc. I was wondering whether s there a way to generate a `torch_geometric.nn.dataset.FakeDataset` with global level features?

Thanks for your help!

**Comment by rusty1s** [comment]
We can definitely think about adding such a feature to `FakeDataset` and `FakeHeteroDataset`. Please let me know if you are interested in contributing this feature.

**Reply by arunppsg** [reply]
Maybe can we add a `kwargs` to __init__ method, where the user can specify the name of the additional attribute (global_features, z or anything) and its shape? I will put up a small PR regarding the same.

**Comment by rusty1s** [comment]
Being worked on in https://github.com/pyg-team/pytorch_geometric/pull/4074.
